src/main/java/org/terasology/rendering/nui/layers/hud/InventoryHud.java

```diff
@@ -69,4 +80,28 @@ public Boolean get() {
             return component != null && component.slot == slot;
         }
     }
+
+    private class TargetSlotBinding implements Binding<Integer> {
+
+        private int offset;
+        private LocalPlayer localPlayer;
+
+        public TargetSlotBinding(int targetSlot, LocalPlayer localPlayer) {
+            this.offset = targetSlot;
+            this.localPlayer = localPlayer;
+        }
+
+        @Override
+        public Integer get() {
+            SelectedInventorySlotComponent component =
+                    localPlayer.getCharacterEntity().getComponent(SelectedInventorySlotComponent.class);
+            return (component.slot + offset) % 10;
+            //return null;
+        }
+
+        @Override
+        public void set(Integer value) {
+
```

Comment:
why is there an empty function here?

Reply:
If this method gets removed, then `TargetSlotBinding` gets an `Class 'TargetSlotBinding' must either be declared abstract or implement abstract method 'set(T)' in 'Binding'` error. 

The two solutions IntelliJ suggests, are:
- To Implement Methods (the above solution, but we get an empty function)
- To make `TargetSlotBinding` abstract

Reply:
okay, guess that's a Java quirk I have to get used to 🙄